src/values/FunctionValue.js

```diff
@@ -13,25 +13,17 @@ import type { ObjectKind } from "../types.js";
 import type { LexicalEnvironment } from "../environment.js";
 import type { Realm } from "../realm.js";
 import { ObjectValue, NumberValue } from "./index.js";
-import * as t from "babel-types";
-import type { BabelNodeLVal, BabelNodeBlockStatement, BabelNodeSourceLocation } from "babel-types";
 import invariant from "../invariant.js";
 
+/* Abstract base class for all function objects */
 export default class FunctionValue extends ObjectValue {
   constructor(realm: Realm, intrinsicName?: string) {
     super(realm, realm.intrinsics.FunctionPrototype, intrinsicName);
   }
 
   $ConstructorKind: void | string;
```

Comment:
The spec does not have something that corresponds to NativeFunctionValue. It does talk about builtin function objects (http://www.ecma-international.org/ecma-262/7.0/#sec-built-in-function-objects). By the sounds of it, I suspect that it may be OK to just make NativeFunctionValue a subclass of ECMAScriptFunctionValue. Alternatively, you can just add the necessary fields to its class definition as well. Factoring out the commonality between NativeFunctionValue and ECMAScriptFunctionValue into a separate base class is not necessarily helpful.